### PR TITLE
Fix small Explorer bugs about file operations

### LINKED ISSUE #3186: "New -> Module..." menu not saving the new module file to the disk
## Description

**What steps will reproduce the problem?**

1. In the Project explorer or File explorer, right click to pop up the context menu, and then select menu "New -> Module...". 
2. The "New module" dialog shows up for the user to input the module file name, say "a.py". After clicking the Save button on the dialog, the module file "a.py" is not saved to the disk. Also, in the Editor, a new tab is opened for the new module with the default name "untitled0.py", rather than "a.py".
3. If you edit and try to save the new module, the "Save file" dialog pops up and requires to input the module name again.

**What is the expected output? What do you see instead?**

I would expect that the new module to be saved to the disk, and the Editor shows the new module with the correct name.

**Please provide any additional information below**

It seems this problem exists for other versions of Spyder as well.

## Version and main components

* Spyder Version: 3.0.0dev 
* Python Version: 2.7.11
* Qt Versions: 5.5.1, PyQt5 5.5.1 on Darwin

## Dependencies
```
pyflakes >=0.5.0:  1.0.0 (OK)
pep8 >=0.6      :  1.7.0 (OK)
pygments >=1.6  :  2.1 (OK)
qtconsole >=4.0 :  4.1.1 (OK)
nbconvert >=4.0 :  4.1.0 (OK)
pandas >=0.13.1 :  0.17.1 (OK)
numpy >=1.7     :  1.11.0 (OK)
sphinx >=0.6.6  :  1.3.3 (OK)
rope >=0.9.4    :  0.9.4-1 (OK)
jedi >=0.8.1    :  0.8.1 (OK)
psutil >=0.3    :  3.4.2 (OK)
matplotlib >=1.0:  1.5.1 (OK)
sympy >=0.7.3   :  0.7.6.1 (OK)
pylint >=0.25   :  1.5.2 (OK)

```

### LINKED ISSUE #3187: Spyder crashes after deleting a folder in File explorer
## Description

**What steps will reproduce the problem?**

1. In the File explorer, right click on a folder. In the popped up menu, select "Delete..." and click "Yes" button in the next confirmation dialog.
2. The folder is deleted, and immediately Spyder crashes, with the following error log:

>   File "/Users/shenggao/anaconda/lib/python2.7/site-packages/spyderlib/widgets/explorer.py", line 515, in delete
>     yes_to_all = self.delete_file(fname, multiple, yes_to_all)
>   File "/Users/shenggao/anaconda/lib/python2.7/site-packages/spyderlib/widgets/explorer.py", line 497, in delete_file
>     self.parent_widget.removed_tree.emit(fname)
> AttributeError: 'Explorer' object has no attribute 'removed_tree'
> Abort trap: 6

**What is the expected output? What do you see instead?**

**Please provide any additional information below**

This is the latest dev version on May 26.

## Version and main components

* Spyder Version: 3.0.0dev 
* Python Version: 2.7.11
* Qt Versions: 5.5.1, PyQt5 5.5.1 on Darwin

## Dependencies
```
pyflakes >=0.5.0:  1.0.0 (OK)
pep8 >=0.6      :  1.7.0 (OK)
pygments >=1.6  :  2.1 (OK)
qtconsole >=4.0 :  4.1.1 (OK)
nbconvert >=4.0 :  4.1.0 (OK)
pandas >=0.13.1 :  0.17.1 (OK)
numpy >=1.7     :  1.11.0 (OK)
sphinx >=0.6.6  :  1.3.3 (OK)
rope >=0.9.4    :  0.9.4-1 (OK)
jedi >=0.8.1    :  0.8.1 (OK)
psutil >=0.3    :  3.4.2 (OK)
matplotlib >=1.0:  1.5.1 (OK)
sympy >=0.7.3   :  0.7.6.1 (OK)
pylint >=0.25   :  1.5.2 (OK)

```

### LINKED ISSUE #3188: Spyder crashes after creating a new file from Project/File Explorer
## Description

**What steps will reproduce the problem?**

1. In the Project explorer or File explorer, right click to pop up the context menu, and then select menu "New -> File...". 
2. In the "New file" dialog, input the file name and click the Save button.
3. The new file is saved to the disk, and immediately Spyder crashes, with the following error log:

> Traceback (most recent call last):
>   File "/Users/shenggao/anaconda/lib/python2.7/site-packages/spyderlib/app/spyder.py", line 2536, in open_file
>     self.variableexplorer.import_data(fname)
>   File "/Users/shenggao/anaconda/lib/python2.7/site-packages/spyderlib/plugins/variableexplorer.py", line 169, in import_data
>     nsb.import_data(filename=fname)
> TypeError: import_data() got an unexpected keyword argument 'filename'
> Abort trap: 6

**What is the expected output? What do you see instead?**

**Please provide any additional information below**
This is the latest dev version on May 26.

## Version and main components

* Spyder Version: 3.0.0dev 
* Python Version: 2.7.11
* Qt Versions: 5.5.1, PyQt5 5.5.1 on Darwin

## Dependencies
```
pyflakes >=0.5.0:  1.0.0 (OK)
pep8 >=0.6      :  1.7.0 (OK)
pygments >=1.6  :  2.1 (OK)
qtconsole >=4.0 :  4.1.1 (OK)
nbconvert >=4.0 :  4.1.0 (OK)
pandas >=0.13.1 :  0.17.1 (OK)
numpy >=1.7     :  1.11.0 (OK)
sphinx >=0.6.6  :  1.3.3 (OK)
rope >=0.9.4    :  0.9.4-1 (OK)
jedi >=0.8.1    :  0.8.1 (OK)
psutil >=0.3    :  3.4.2 (OK)
matplotlib >=1.0:  1.5.1 (OK)
sympy >=0.7.3   :  0.7.6.1 (OK)
pylint >=0.25   :  1.5.2 (OK)

```

### PR DESCRIPTION
**Pull request by ShenggaoZhu**
Fixes #3186
Fixes #3187
Fixes #3188

### CODE REVIEW

**Comment by ccordoba12** [comment]
@ShenggaoZhu, thanks a lot for helping us to improve Spyder! It's really appreciated :-)

Could you please merge with master (or rebase your branch, as you prefer :-) to make our AppVeyor tests to pass? Thanks!

**Comment by ccordoba12** [comment]
This has now conflicts with master, sorry. Please merge again with master to fix them :-)

**Comment by ShenggaoZhu** [comment]
Thanks for your guidance! This is my first PR :-) I have rebased my commits on the latest master branch. Looks clean now. Could you please check if you can merge it now?

File: spyderlib/plugins/editor.py [hunk 1]
```diff
@@ -1521,7 +1521,7 @@ def _clone_file_everywhere(self, finfo):
             editor = editorstack.clone_editor_from(finfo, set_current=False)
             self.register_widget_shortcuts("Editor", editor)
     
-    @Slot()
+#    @Slot(str)
```
Comment (ccordoba12) [inline spyderlib/plugins/editor.py]: This is an error. Please leave this as

```python
@Slot()
@Slot(str)
def new(self, fname=None, editorstack=None, text=None):
```

We need those two slot signatures for the `new` method to work correctly.

Comment (ccordoba12) [inline spyderlib/plugins/editor.py]: We're in a bit of a hurry, so I'll do the fix myself in master :-)

Thanks a lot for your contribution, I verified that it does the right thing!

**Pull request merged**